nuxeo-distribution/nuxeo-launcher/src/main/java/org/nuxeo/launcher/NuxeoLauncher.java

```diff
@@ -1262,6 +1277,94 @@ public void run() {
         }
     }
 
+    protected boolean registerRemoteInstance() throws IOException, ConfigurationException, PackageException {
+        String username;
+        String password;
+        String project;
+        String description = "";
+        NuxeoClientInstanceType type = null;
+        if (System.console() != null) {
+            username = LauncherRegisterPromptHelper.promptUsername();
+            password = LauncherRegisterPromptHelper.promptPassword();
+            List<ConnectProject> projs = getConnectRegisterationBroker().getAvailableProjects(username, password);
+            project = LauncherRegisterPromptHelper.promptProjectId(projs);
+            type = LauncherRegisterPromptHelper.promptInstanceType();
+            description = LauncherRegisterPromptHelper.promptDescription();
+        } else {
+            String[] input = new String(IOUtils.toByteArray(System.in)).split("\n");
+            if (input.length < 4) {
+                throw new IOException("Wrong number of arguments.");
+            }
+            username = input[0];
+            password = input[1];
+            project = input[2];
+            type = NuxeoClientInstanceType.fromString(input[3]);
+            if (input.length > 3) {
+                description = StringUtils.join(Arrays.copyOfRange(input, 4, input.length), "\n");
+            }
+        }
+
+        if (type == null) {
+            type = NuxeoClientInstanceType.DEV;
+        }
+
+        getConnectRegisterationBroker().registerRemote(username, password, project, type, description);
+        log.info("Your server is correctly registered to: " + username);
+        return true;
+    }
+
+    protected boolean registerTrial() throws IOException, ConfigurationException, PackageException {
+        getConnectBroker(); // Initialize NuxeoConnectClient Oo
+        CommandInfo commandInfo = cset.newCommandInfo("register-trial");
+
+        Map<String, String> registration = new HashMap<>();
+        if (System.console() != null) {
+            registration.put("email", LauncherRegisterPromptHelper.promptMail());
+            registration.put("company", LauncherRegisterPromptHelper.promptCompany());
+
+            String pwd = LauncherRegisterPromptHelper.promptPassword(true);
+            registration.put("password", pwd);
+            registration.put("password_verif", pwd);
+
+            registration.put("connectreg:projectName", LauncherRegisterPromptHelper.promptProjectName());
+            registration.put("description", LauncherRegisterPromptHelper.promptDescription());
+        } else {
+            String[] input = new String(IOUtils.toByteArray(System.in)).split("\n");
```

Comment:
Should then probably check `input` length and display error messages if malformed.